fuel/datasets/base.py

```diff
@@ -28,10 +29,6 @@ class Dataset(object):
         'targets')`` for MNIST (regardless of which data the data stream
         actually requests). Any implementation of a dataset should set this
         attribute on the class (or at least before calling ``super``).
-    default_iteration_scheme : :class:`.IterationScheme`, optional
```

Comment:
Yes, `DataStream(dataset)` only works on datasets which don't need an iteration scheme, that's the idea. Because I'm not convinced `ExampleScheme` is a reasonable default. I almost never use that, which is kind of my point: The sensible default completely depends on the person. If you are using lengthy pipelines that perform pre-processing on an example-level, `ExampleScheme` is a reasonable default. However, that default would crash our most basic tutorial simply because most people use minibatches and expect 2D matrices, which would make the current `SequentialScheme` a better default. Doesn't the fact that you are suggesting switching defaults already show that the choice is ambiguous?

Also, under the banner of "explicit is better than implicit" at least `DataStream(dataset, iteration_scheme=SequentialExamples(len(dataset))` makes clear what to expect while `dataset.get_default_stream()` leaves everyone guessing; will we get minibatches of size 1, individual examples, or something else? Changing the default later on will break things needlessly.

Then there's the last, but slightly OCD argument: The default scheme makes all the dataset files depend on the schemes. I like the idea of there not being any dependency.